rust/src/adapter/otel_formatter/mod.rs

```diff
@@ -1,234 +1,121 @@
-        None => s.serialize_str(""),
-    }
-}
-
-#[derive(Default, Debug, Clone, PartialEq, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct Value2 {
-    pub string_value: String,
-}
-
-#[derive(Default, Debug, Clone, PartialEq, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct Status {}
-
 impl OtelFormatter {
-    pub fn new() -> OtelFormatter {
+    pub fn new(
+        spans: Vec<opentelemetry::proto::trace::v1::Span>,
+        service_name: String,
+    ) -> OtelFormatter {
         OtelFormatter {
-            resource_spans: Vec::new(),
-        }
-    }
-
-    pub fn add_resource_span(&mut self, rs: ResourceSpan) {
-        self.resource_spans.push(rs)
-    }
-}
-
-impl ResourceSpan {
-    pub fn new() -> ResourceSpan {
-        ResourceSpan {
-            scope_spans: vec![],
-            resource: Resource {
-                attributes: Vec::new(),
+            traces_data: opentelemetry::proto::trace::v1::TracesData {
+                resource_spans: new_resource_spans(spans, service_name),
             },
         }
     }
 
-    pub fn add_attribute(mut self, key: String, value: String) -> ResourceSpan {
-        let attribute = Attribute {
-            key,
-            value: Value {
-                string_value: Some(value),
-                int_value: None,
-            },
-        };
-        self.resource.attributes.push(attribute);
-        self
-    }
-
-    pub fn add_spans(&mut self, spans: Vec<Span>) {
-        self.scope_spans = vec![ScopeSpan {
-            scope: Scope {
-                name: "event".to_string(),
-            },
-            spans,
-        }]
-    }
-}
-
-impl Span {
-    pub fn new(
+    pub fn new_span(
         trace_id: String,
-        parent_id: Option<String>,
+        parent_id: Vec<u8>,
         name: String,
         start_time: SystemTime,
         end_time: SystemTime,
-    ) -> Span {
-        Span {
-            trace_id,
-            span_id: new_span_id().to_hex_8(),
+    ) -> opentelemetry::proto::trace::v1::Span {
+        opentelemetry::proto::trace::v1::Span {
+            trace_id: trace_id.into(),
+            span_id: new_span_id().to_hex_8().as_bytes().to_vec(),
             parent_span_id: parent_id,
             name,
             kind: 1,
             start_time_unix_nano: start_time
                 .duration_since(SystemTime::UNIX_EPOCH)
                 .unwrap()
-                .as_nanos(),
+                .as_nanos() as u64,
             end_time_unix_nano: end_time
                 .duration_since(SystemTime::UNIX_EPOCH)
                 .unwrap()
-                .as_nanos(),
+                .as_nanos() as u64,
             attributes: Vec::new(),
             dropped_attributes_count: 0,
             dropped_events_count: 0,
             dropped_links_count: 0,
-            status: Status {},
+            status: None,
+            events: vec![],
+            links: vec![],
+            trace_state: "".into(),
```

Comment:
no need to update in the PR, but for future cases like this you can pass `..Default::default()` as the last field in a struct literal as long as the Default trait is implemented. This will fill in the rest of the fields not explicitly included with their default values (like you've done manually).